### PR TITLE
load kubeconfig from custom environment variables 

### PR DESCRIPTION
**Pull request by goenning**
Signed-off-by: goenning <me@goenning.net>

## Motivation
I'd like to support loading kubeconfig files from a non-standard environment variable name.

## Solution
I first tried to copy this function over to my project, but Kubeconfig's merge fn is private. We could make it public, but I thought it might be beneficial to make have this small additional function on kube-rs

### CODE REVIEW

File: kube-client/src/config/file_config.rs [hunk 1]
```diff
@@ -350,7 +350,17 @@ impl Kubeconfig {
...
+    }
+
+    /// Create `Kubeconfig` from a given environment variable.
+    /// Supports list of files to be merged.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the environment variable value contains the NUL character.
+    pub fn from_custom_env(env_name: &str) -> Result<Option<Self>, KubeconfigError> {
+        match std::env::var_os(env_name) {
```
Comment (clux) [inline kube-client/src/config/file_config.rs:363]: As much as it feels like a rare use-case to have people use different evars, **this way** of exporting two sides of the functionality in `from_env` that you have here, is probably the best split.

Tried to think of a few other splits where `::from_env` would parse the KUBECONFIG first, and maybe splits the paths, then maybe collects (and maybe even reads the files), before then calling a merger fn; but none of those options feel like they are actually giving us a useful split IO/non-IO split without making a very ugly interface.

That said, I'm kind of more positive on exporting the `merge` fn here than to have `kube` read a different non-standard evar as that feels very unexpected for kube to do (and something i expect to be an extremely rare use-case - unless I am missing something).

Comment (goenning) [reply kube-client/src/config/file_config.rs:374]: 100% understandable, it's a rare use-case indeed. Exporting merge would also work for me, so I created this other PR #1100

**Comment by clux** [comment]
Replaced by #1100

File: kube-client/src/config/file_config.rs (hunk 1 above)
Comment (clux) [reply kube-client/src/config/file_config.rs:374]: Appreciate it. Have merged this and closed this PR. It's nicer to have fns that export documented parts of kubernetes algorithms be public.

**Pull request closed without merging**